### PR TITLE
Add a code fix that applies ProducesResponseTypeAttributes

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Analyzers/AddResponseTypeAttributeCodeFixAction.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+                statusCodes.Add(metadata.IsDefaultResponse ? 200 : metadata.StatusCode);
+            }
+
+            return statusCodes;
+        }
+
+        private static AttributeSyntax CreateProducesResponseTypeAttribute(int statusCode)
+        {
+            return SyntaxFactory.Attribute(
```
Comment (khellang) [inline src/Microsoft.AspNetCore.Mvc.Analyzers/AddResponseTypeAttributeCodeFixAction.cs]: `using static`?

**Comment by pranavkm** [comment]
@dotnet-bot test Windows Release x64 Build please

File: src/Microsoft.AspNetCore.Mvc.Analyzers/AddResponseTypeAttributeCodeFixAction.cs (hunk 1 above)
Comment (pranavkm) [reply src/Microsoft.AspNetCore.Mvc.Analyzers/AddResponseTypeAttributeCodeFixAction.cs]: I used this in another version of this change - it's less typing, but I ended up typing `SyntaxFactory.` to look up what methods it offers far too often.

File: test/Mvc.Analyzers.Test/TestFiles/AddResponseTypeAttributeCodeFixProviderIntegrationTest/CodeFixWithConventionMethodAddsMissingStatusCodes.Output.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿namespace Microsoft.AspNetCore.Mvc.Analyzers._OUTPUT_
+{
+    [ApiController]
+    [Route("[controller]/[action]")]
+    public class CodeFixWithConventionMethodAddsMissingStatusCodes : ControllerBase
+    {
+        [ApiConventionMethod(typeof(DefaultApiConventions), nameof(DefaultApiConventions.Find))]
```
Comment (pranavkm) [inline test/Mvc.Analyzers.Test/TestFiles/AddResponseTypeAttributeCodeFixProviderIntegrationTest/CodeFixWithConventionMethodAddsMissingStatusCodes.Output.cs]: @rynowak thoughts on leaving this around?

Comment (rynowak) [reply test/Mvc.Analyzers.Test/TestFiles/AddResponseTypeAttributeCodeFixProviderIntegrationTest/CodeFixWithConventionMethodAddsMissingStatusCodes.Output.cs]: It should remove this, it's not meaningful

File: test/Mvc.Analyzers.Test/AddResponseTypeAttributeCodeFixProviderIntegrationTest.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            Assert.Equal(expectedOutput, actualOutput, ignoreLineEndingDifferences: true);
+        }
+
+        private Project GetProject(string testMethod)
+        {
+            var testSource = Read(testMethod + ".Input");
+            return DiagnosticProject.Create(GetType().Assembly, new[] { testSource });
+        }
+
+        private string Read(string fileName)
```
Comment (NTaylorMullen) [inline test/Mvc.Analyzers.Test/AddResponseTypeAttributeCodeFixProviderIntegrationTest.cs]: Will replacing baselines be a thing for this? If not, then it's good as is.

Comment (pranavkm) [reply test/Mvc.Analyzers.Test/AddResponseTypeAttributeCodeFixProviderIntegrationTest.cs]: So the plan is to ship this and revise this based on preview1 feedback. It's sort of intentionally lightly (unit) tested to see if this approach sticks \ if we need to take a different approach to providing these code fixes etc. I didn't want to too much infrastructure to support it at this point. But yeah, we would want to do the generate base lines thing at some point

**Pull request closed without merging**